### DISCUSSION TITLE
GitHub API fetches blocked by proxy?

### DISCUSSION THREAD

**Discussion by MetRonnie** [github/vscode-github-actions#127] [Q&A]
Any ideas for how to figure out what's going wrong here? Could it be caused by workplace proxy server?

![image](https://github.com/github/vscode-github-actions/assets/61982285/a8184468-1ee5-4635-b642-c13361cce1c6)

From the logs for GitHub Actions Language Server:

> [Error - 10:48:06 AM] Failed to fetch action metadata for actions/checkout/v3: 'request to https://api.github.com/repos/actions/checkout/contents/action.yml?ref=v3 failed, reason: connect ETIMEDOUT <some_ip_address>:443'

and
```
Failure to retrieve username:  Ms [HttpError]: request to https://api.github.com/user failed, reason: connect ETIMEDOUT <some_ip_address>:443
    at ~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2785024
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2856983
    at async wa.get (~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2866891)
    at async ~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2856918
    at async la (~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2856885)
    at async ~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2860689
    at async Object.getContext (~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2860645)
    at async oi (~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2733949)
    at async gn (~/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/server-node.js:2:2759192) {
  status: 500,
  request: {
    method: 'GET',
    url: 'https://api.github.com/user',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'VS Code GitHub Actions (0.25.4) octokit-rest.js/19.0.7 octokit-core.js/4.1.0 Node.js/16.14.2 (linux; x64)',
      authorization: 'token [REDACTED]'
    },
    request: { hook: [Function: bound bound e] }
  }
```

**Comment by nschonni** [comment]
Maybe a duplicate of https://github.com/github/vscode-github-actions/issues/39

**Comment by KetchupOnMyKetchup** [comment]
The unable to resolve action `actions/checkout@v3` pop up I don't think is related to being behind a proxy.  I am able to repro it as well on my Windows machine: 
![image](https://github.com/github/vscode-github-actions/assets/7976517/558eaf33-f1f9-40ea-ac97-9f3f67acb179)

But using WSL it works:
![image](https://github.com/github/vscode-github-actions/assets/7976517/4b7dca2d-3a8e-4264-b69a-035ccd9120f6)

And we just checked and it works as well on Mac. 
![image](https://github.com/github/vscode-github-actions/assets/7976517/8993bfdf-edf8-4a02-b518-ab7c8473a022)

I think it is likely a bug related to pathing with Windows. 

Just to verify, are you on Windows as well? If so, could you please submit a bug to us with this?

**Reply by MetRonnie** [reply]
I am on Linux x64 3.10.0-1160.88.1.el7.x86_64

VSCode 1.77.0

Edit: but also get the same problem on Windows at the same workplace

**Reply by KetchupOnMyKetchup** [reply]
Hmm ok good to know about the OS, thank you for that info! Could you please submit this as a bug for us so we can get it prioritized? Thank you!

**Reply by MetRonnie** [reply]
Done: https://github.com/github/vscode-github-actions/issues/143

**Reply by romeopeter** [reply]
Hey, where you able to find the fix to the problem? I'm currently have the same issue and I'm yet to find a fix to it.
